hack/run-tests.sh

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -euo pipefail
+
+SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+color_red="$(tput setaf 1)"
+color_green="$(tput setaf 2)"
+color_blue="$(tput setaf 4)"
+color_reset="$(tput sgr0)"
+
+print_with_color() {
+      echo "${1}${*:2}${color_reset}"
+}
+
+print_status() {
+  if [[ $? == 0 ]]; then
```

Comment:
this $? capturing here sounds very fragile.

Reply:
I've extracted the status into a variable and added a comment. Do you think that's enough?

Reply:
Sure yeah I think it works, although if I go edit 

```
local result="$?"
```

with my VSCode which has `shellcheck` installed, it'll tell me to separate it due to https://github.com/koalaman/shellcheck/wiki/SC2155, such as:


```
local result
result=$?
```

which again breaks, but keeps the warning.

----

What I recommend here instead of capturing $? and setting up trap is one of these options:

1. don't do anything fancy, you already have `set -e`, let the scripts fail, because they will fail with an actionable error message. + we don't need SUCCESS message.
2. just call `print_fail` when something fails like `./cmd || print_fail`, and have `print_fail` to do just `exit 1` with a RED message.

Reply:
My shellcheck (v0.6.0) does not warn here. Would the error go away with
```
local -r result="$?"
```
The description from the link sounds as if this is a false positive anyway (there can be no return value, as `$?` is simply a variable).